scripts/test_usb_scan.py

```diff
@@ -0,0 +1,393 @@
+                }
+            }
+        ]
+        results = [
+            {
+                "product-desc": "",
+                "product-id": "0302",
+                "description": "Feitian Technologies, Inc.",
+                "vendor-desc": "Feitian Technologies, Inc.",
+                "version": "1.10",
+                "vendor-id": "096e",
+                "path": "1-2.1",
+                "serial": ""
+            }
+        ]
+        self.test_usb_common(devices, interfaces, results)
+
+    def test_usb_dongle_unbinded(self):
+        devices = [
+            {
+                "name": "1-2",
+                "props": {
+                    "ID_VENDOR_FROM_DATABASE": "Feitian Technologies, Inc."
+                },
+                "attrs": {
+                    "idVendor": "096e",
+                    "bNumInterfaces": "",
+                    "bConfigurationValue": "",
+                    "bcdDevice": "010a",
+                    "version": " 1.10",
+                    "idProduct": "0302",
+                    "bDeviceClass": "00",
+                }
+            }
+        ]
+        interfaces = [
+        ]
+        results = [
+        ]
+        self.test_usb_common(devices, interfaces, results)
+
+    def test_usb_keyboard(self):
+        devices = [
+            {
+                "name": "1-2",
+                "props": {
+                    "ID_VENDOR_FROM_DATABASE": "Dell Computer Corp."
+                },
+                "attrs": {
+                    "idVendor": "413c",
+                    "bNumInterfaces": " 2",
+                    "bConfigurationValue": "1",
+                    "bcdDevice": "0110",
+                    "version": " 2.00",
+                    "idProduct": "2113",
+                    "bDeviceClass": "00",
+                }
+            }
+        ]
+        interfaces = [
+            {
+                "name": "1-2:1.0",
+                "attrs": {
+                    "bInterfaceClass": "03",
+                    "bInterfaceSubClass": "01",
+                    "bInterfaceProtocol": "01",
+                    "bInterfaceNumber": "00",
+                }
+            },
+            {
+                "name": "1-2:1.1",
+                "attrs": {
+                    "bInterfaceClass": "03",
+                    "bInterfaceSubClass": "00",
+                    "bInterfaceProtocol": "00",
+                    "bInterfaceNumber": "01",
+                }
+            }
+        ]
+        results = [
+        ]
+        self.test_usb_common(devices, interfaces, results)
+
+    def test_usb_config_missing(self):
+        self.test_usb_exit([], [], [], "not_exit.conf")
+
+    @nottest
+    def test_usb_config_error_common(self, content):
+        path = os.path.join(self.work_dir, "usb-policy.conf")
+        with open(path, "w") as f:
+            f.write(content)
+        self.test_usb_exit([], [], [], path)
+
+    def test_usb_config_error_01(self):
```

Comment:
This is just a stylistic thing. Here and below, give meaningful names to the error functions. For example, in this case, `test_usb_config_error_unexpected_characters` and in the next one `test_usb_config_error_duplicated_key`, and so on

Reply:
done.